### PR TITLE
7.x deprecated dUseCIEColor

### PR DESCRIPTION
**Pull request by IAMlKeno**
**ISLANDORA-2432**: (https://jira.duraspace.org/browse/ISLANDORA-2432)

# What does this Pull Request do?
This pull request with allow the module to handle Ghostscript versions lower than 9.11, where the switch "-dUseCIEColor" is recommended not be used. See https://ghostscript.com/pipermail/gs-devel/2014-July/009693.html

# What's new?
When generating the PDF/A datastream, the -dUseCIEColor switch will be used based on the version of Ghostscript being used on the server. If the version is 9.11 or greater, the switch is not used.

# How should this be tested?
-- Using a version of Ghostscript < 9.11 and a version greater --
* Ensure that Create PDF/A archival derivative from PDF is enabled in PDF configuration. 
* Ingest a PDF that is composed mostly of images. 
* The object should be ingested normally and the PDFA datastream generated successfully.
* The view of the object should function as normal

# Interested parties
@Islandora/7-x-1-x-committers
@jonathangreen

### CODE REVIEW

**Comment by IAMlKeno** [comment]
The checks should pass when this pull request is merged 
https://github.com/Islandora/islandora_solution_pack_pdf/pull/139

File: includes/derivatives.inc
Comment (jonathangreen) [inline includes/derivatives.inc]: Should probably do a more sophisticated version comparison here, in case the version happens to return a point release something like this: `9.11.2`. In this case it won't cast to a number... and then you are doing `>=` comparison on strings...

Maybe use something like this instead: 
https://php.net/version_compare

File: includes/derivatives.inc
Comment (jonathangreen) [inline includes/derivatives.inc]: How much time does this add to the derivative creation process to run this version command every time? I think maybe we should do some performance testing on this. If it adds significant time, maybe we could have it as an admin option, or cache the result of this for some time?

**Review by jonathangreen — CHANGES_REQUESTED** [review]
Have a couple concerns in the comments.

File: includes/derivatives.inc
Comment (IAMlKeno) [reply includes/derivatives.inc]: Using it as an admin option seems to a good bet

File: includes/derivatives.inc
Comment (IAMlKeno) [reply includes/derivatives.inc]: Eliminated the need for this check by making this an admin option

File: includes/derivatives.inc
Comment (willtp87) [reply includes/derivatives.inc]: Should probably have an update hook to set the variable to the correct value.

Comment (IAMlKeno) [reply includes/derivatives.inc]: Thanks for pointing that out! Adding in the .install file

File: islandora_pdf.install
Comment (mghughes) [inline islandora_pdf.install]: I think this should be numbered 7002 if we're following our usual conventions for numbering update hooks

**Review by mghughes — APPROVED** [review]

**Comment by IAMlKeno** [comment]
ping @jonathangreen, are the changes satisfactory?

**Comment by jonathangreen** [comment]
Code changes look good to me! 👍🏻

I guess this needs to get sorted before we can merge:
https://jira.duraspace.org/browse/ISLANDORA-2427

**Comment by nhart** [comment]
@jonathangreen  can we get this merged before the deprecated bits are figured out? This has been sitting for a while.

**Comment by jonathangreen** [comment]
I agree its been waiting for awhile. I don't feel comfortable merging when travis is red without some input from other @Islandora/7-x-1-x-committers or our fearless leader @dannylamb providing some feedback that its okay to merge PRs with travis failing right now.

**Comment by nhart** [comment]
Correct me if I'm wrong, but the line it's complaining about isn't part of this PR and will no doubt we part of a large find and replace of all islandora_deprecated calls, no?

**Review by jonathangreen — APPROVED** [review]
LGTM pending travis

**Comment by jonathangreen** [comment]
@nhart I approved the PR, and I'm not standing in the way of anyone merging, but I don't personally want to hit the button until there is some consensus about those changes and that we can just ignore travis for now, until we fix it. If any other @Islandora/7-x-1-x-committers is braver then I feel free to merge or provide some feedback that we are good to merge / not merge.

**Comment by DiegoPino** [comment]
@nhart quick solution, since most committers are working on 8 would be to add an ignore coding standards php doc statement 
```PHP
// @codingStandardsIgnoreStart
```

,around the offending `trigger_error(filter_xss($message), E_USER_DEPRECATED);`
Does that sound possible? Its already approved so merge could happen before 24 hours. I also don't feel comfortable pressing the merge button if not.

**Comment by whikloj** [comment]
Sorry folks I think it was on my todo to ignore that trigger error line. I have added that to #139 so hopefully that will unstick this.

**Comment by DiegoPino** [comment]
Restarting Travis. maybe it will take the change?

**Comment by DiegoPino** [comment]
This pull is failing just because the world is a dark, cold place.... can anyone give travis a look? Not even related to this pull anymore. Restarted it 3x

**Comment by whikloj** [comment]
Sadly I think @IAMlKeno will need to rebase this branch on 7.x to pick up the changes to the `travis.yml` file, then 5.3.3 will be allowed to fail.

**Comment by nhart** [comment]
thank you @IAMlKeno!

**Comment by jonathangreen** [comment]
Merged! Thanks @IAMlKeno for working through this one and apologies for holding it up. I think we need to all work together to make sure our CI is running though, so I've super happy with the outcome to see travis green when merging this. I'll get off my soap box and go back to work now 😄

**Comment by IAMlKeno** [comment]
Thanks, that sounds good @jonathangreen

**Pull request merged**
